esx_service/vmdk_ops.py

```diff
@@ -786,9 +786,23 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     vm_datastore_url = vmdk_utils.get_datastore_url_from_config_path(config_path)
     vm_datastore = get_datastore_name(vm_datastore_url)
 
+    # Detach is implicity allowed if a disk has been attached
+    # Do not find tenant or privileges; just detach the disk
+    if cmd == "detach":
+        with lockManager.get_lock(vm_uuid):
+            response = detachVMDK(full_vol_name, vm_uuid)
+
```

Comment:
nit (can be ignored): generally I'd prefer to have a simple dict-driven code, with something like  {"attach": ["auth_needed": True, "func": "code", ,,} ...}  to drive the execution for commands. (this is the same note as the above reply to @shaominchen ).